tensorflow_recommenders_addons/dynamic_embedding/core/kernels/rocksdb_table_op.cc

```diff
@@ -0,0 +1,1318 @@
+        static DBWrapperRegistry &instance() {
+          static DBWrapperRegistry instance;
+          return instance;
+        }
+
+      private:
+        DBWrapperRegistry() = default;
+
+        ~DBWrapperRegistry() = default;
+
+      public:
+        std::shared_ptr<DBWrapper> connect(
+          const std::string &databasePath, const bool &readOnly
+        ) {
+          mutex_lock guard(lock);
+
+          // Try to find database, or open it if it is not open yet.
+          std::shared_ptr<DBWrapper> db;
+          auto pos = wrappers.find(databasePath);
+          if (pos != wrappers.end()) {
+            db = pos->second.lock();
+          }
+          else {
+            db.reset(new DBWrapper(databasePath, readOnly), deleter);
+            wrappers[databasePath] = db;
+          }
+
+          // Suicide, if the desired access level is below the available access level.
+          if (readOnly < db->readOnly()) {
+            throw std::runtime_error("Cannot simultaneously open database in read + write mode.");
+          }
+
+          return db;
+        }
+
+      private:
+        static void deleter(DBWrapper *wrapper) {
+          static std::default_delete<DBWrapper> defaultDeleter;
+
+          DBWrapperRegistry &registry = instance();
+          const std::string path = wrapper->path();
+
+          // Make sure we are alone.
+          mutex_lock guard(registry.lock);
+
+          // Destroy the wrapper.
+          defaultDeleter(wrapper);
+          // LOG(INFO) << "Database wrapper " << path << " has been deleted.";
+
+          // Locate the corresponding weak_ptr and evict it.
+          auto pos = registry.wrappers.find(path);
+          if (pos == registry.wrappers.end()) {
+            LOG(ERROR) << "Unknown database wrapper. How?";
+          }
+          else if (pos->second.expired()) {
+            registry.wrappers.erase(pos);
+            // LOG(INFO) << "Database wrapper " << path << " evicted.";
+          }
+          else {
+            LOG(ERROR) << "Registry is in an inconsistent state. This is very bad...";
+          }
+        }
+
+      private:
+        mutex lock;
+        std::unordered_map<std::string, std::weak_ptr<DBWrapper>> wrappers;
+      };
+
+      template<class K, class V>
+      class RocksDBTableOfTensors final : public PersistentStorageLookupInterface {
+      public:
+        /* --- BASE INTERFACE ------------------------------------------------------------------- */
+        RocksDBTableOfTensors(OpKernelContext *ctx, OpKernel *kernel)
+        : readOnly(false), estimateSize(false), dirtyCount(0) {
+          OP_REQUIRES_OK(ctx, GetNodeAttr(kernel->def(), "value_shape", &valueShape));
+          OP_REQUIRES(ctx, TensorShapeUtils::IsVector(valueShape), errors::InvalidArgument(
+            "Default value must be a vector, got shape ", valueShape.DebugString()
+          ));
+
+          OP_REQUIRES_OK(ctx, GetNodeAttr(kernel->def(), "database_path", &databasePath));
```

Comment:
@rhdong @MoFHeKa The JSON method employed by HeKa is of course doable. I thought so far that this was a last resort measure because of the many configurations the Redis implementation has to support. Style-wise it felt to me rather foreign and almost like a breaking change to introduce parameters via an external JSON file. Doesn't mean I don't want to do it. Just want to confirm that that is the way you want it to be. Please confirm